### PR TITLE
feat: generate SQL table schema from JSON Schemas [1/n]

### PR DESCRIPTION
**Pull request by gmaclennan**
Generate drizzle-orm SQLite table schemas from Mapeo types & JSONSchemas.
(Currently uses unpublished version of `@mapeo/schema` that exports JSONSchemas with expanded $refs)

### CODE REVIEW

**Comment by gmaclennan** [comment]
Now finished and ready for review, but awaiting publishing of new version of `@mapeo/schema`.

Will be stacking other PRs on this, working on #139

File: lib/schema/schema-to-drizzle.js [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+/**
+Convert optional properties to nullable
+@template {object} T
+@typedef {{ [K in keyof T]-?: undefined extends T[K] ? T[K] | null : T[K] }} OptionalToNull
+ */
+
+/**
+Convert a readonly array/object to writeable
+@template T
+@typedef {{ -readonly [P in keyof T]: T[P] }} Writable
```
Comment (achou11) [inline lib/schema/schema-to-drizzle.js:12]: out of curiosity, what are the hyphens in the types?

File: lib/schema/schema-to-drizzle.js [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+
+/**
+ * Check if a value is an array of strings of length at least one
+ * @param {any} value
+ * @returns {value is [string, ...string[]]}
+ */
+function isStringArray(value) {
+  return (
+    isArray(value) &&
+    value.every((v) => typeof v === 'string') &&
```
Comment (achou11) [inline lib/schema/schema-to-drizzle.js:299]: was going to suggest prematurely optimizing this using `Array.some`, but the length `value` will be small here like 99% of the time right?

**Review by achou11 — APPROVED** [review]
overall looks good! curious about what else to add to the tests, if anything

File: lib/schema/schema-to-drizzle.js (hunk 1 above)
Comment (gmaclennan) [reply lib/schema/schema-to-drizzle.js:12]: They are called "[mapping modifiers](https://www.typescriptlang.org/docs/handbook/2/mapped-types.html#mapping-modifiers)" and can be used to remove readonly from a type.

File: lib/schema/schema-to-drizzle.js (hunk 2 above)
Comment (gmaclennan) [reply lib/schema/schema-to-drizzle.js:299]: Yes, will be small e.g. max will be the number of enum values. Also this only runs once on app startup so it's not something that needs to be fast.

**Comment by gmaclennan** [comment]
> overall looks good! curious about what else to add to the tests, if anything

I think more type tests would be good, but probably make more sense to be end-to-end tests rather than unit-testing just this function.

File: lib/schema/schema-to-drizzle.js (hunk 1 above)
Comment (achou11) [reply lib/schema/schema-to-drizzle.js:12]: oh wow, very cool!

**Comment by gmaclennan** [comment]
Merged in #149

**Pull request closed without merging**